### ISSUE TITLE
5.0.0-beta in class ShardingSphereAlgorithmBeanRegistry some spring bean would cause java.lang.NullPointerException

### ISSUE THREAD

**Issue by Erzhoumu** [apache/shardingsphere#11008]
hi,  starting a spring-boot project, when parse the sharing:rule element
The class ShardingSphereAlgorithmBeanRegistry in line 25

```
public static Map<String, RuntimeBeanReference> getAlgorithmBeanReferences(ParserContext parserContext, Class<? extends ShardingSphereAlgorithmFactoryBean<?>> algorithmFactoryBeanClass) {
        String[] beanDefinitionNames = parserContext.getRegistry().getBeanDefinitionNames();
        String algorithmFactoryBeanClassName = algorithmFactoryBeanClass.getName();
        Map<String, RuntimeBeanReference> result = new ManagedMap(beanDefinitionNames.length);
        String[] var5 = beanDefinitionNames;
        int var6 = beanDefinitionNames.length;

        for(int var7 = 0; var7 < var6; ++var7) {
            String each = var5[var7];
            if (parserContext.getRegistry().getBeanDefinition(each).getBeanClassName().equals(algorithmFactoryBeanClassName)) {
                result.put(each, new RuntimeBeanReference(each));
            }
        }

        return result;
    }
```

The code may cause NullPointerException, when 'each' is  "org.springframework.context.annotation.internalAsyncAnnotationProcessor"  or "org.springframework.context.annotation.internalScheduledAnnotationProcessor" 

```
  if (parserContext.getRegistry().getBeanDefinition(each).getBeanClassName().equals(algorithmFactoryBeanClassName)) {
                result.put(each, new RuntimeBeanReference(each));
            }
```
I think foreach all beanDefinitionNames is not a good choice, and some lazy-init bean may cause exception.
my spring-boot version is <version>2.1.1.RELEASE</version>

**Comment by tristaZero** [comment]
@Erzhoumu Thanks for your feedback.
It looks like a bug?
Would you like to raise a PR to improve it? FYI, [How to raise a PR](https://shardingsphere.apache.org/community/cn/contribute/contributor/).

**Comment by sandynz** [comment]
I checked on master branch code, it has been fixed by #11702. You could try version `5.0.0`.
